memtable.cc

```diff
@@ -211,8 +212,14 @@ memtable::find_or_create_partition(const dht::decorated_key& key) {
         memtable_entry* entry = current_allocator().construct<memtable_entry>(
             _schema, dht::decorated_key(key), mutation_partition(_schema));
         partitions.insert_before(i, *entry);
+        if (_table_stats) {
+            ++_table_stats->partition_creations;
+        }
         return entry->partition();
```

Comment:
Better to force the caller to supply a stats object than to add conditionals everywhere. Statistics should be cheap.

Reply:
Done. On memtable construction, if no `table_stats` are provided, a dummy `thread_local` object will be supplied, similarly to the `dirty_memory_manager` object.

Reply:
Yes, that's fine.